sql/analyzer/indexes.go

```diff
@@ -258,6 +260,43 @@ func getIndexes(
 		}
 
 		return result, nil
+	case *expression.Like:
```

Comment:
simplification should maybe live separately from applying indexes. Is there any reason we shouldn't just convert LIKE(x,a) to BETWEEN(x,a,a) in all plans?